mlir/lib/Ion/Transforms/oqd_database_managers.h

```diff
@@ -0,0 +1,36 @@
+#include <dlfcn.h>
+#include <iostream>
+
+//#include <nanobind/nanobind.h>
+//#include <nanobind/eval.h>
+//#include <nanobind/stl/string.h>
+
+using namespace std;
+
+void oqd_hi(){
+	std::string path_to_this_file = __FILE__;
+    std::cout << "Current working file: " << path_to_this_file << std::endl;
+    std::string to_remove = "mlir/lib/Ion/Transforms/./oqd_database_managers.h";
+
+    size_t pos = path_to_this_file.find(to_remove);
+
+    std::string manager_so_location = path_to_this_file.replace(pos, to_remove.length(),
+        	"frontend/catalyst/third_party/oqd/oqd_database_managers.so");
+
+
+    std::cout << ".so location: " << manager_so_location << std::endl;
+
+
+	void *handle;
+    void (*libfunc)();
+
+    handle = dlopen((manager_so_location).c_str(), RTLD_LAZY);
```

Comment:
I just realized that perhaps the linking against the python api library shouldn't happen in the cmake, but should happen here in the pass itself, since we are linking dynamically ("DL"). The linking should happen at runtime of the pass instead of compile time of the pass.

The problem now becomes, where is the `.so` for the python api library?

Reply:
Likely the `.so` is unavailable. In that case, just `#include <Python.h>` here in the pass?